vapoursynth/src/node/mod.rs

```diff
@@ -104,22 +105,40 @@ impl<'core> Node<'core> {
     }
 
     /// Generates a frame directly.
+    /// Returns an error if the requested frame number is beyond the `VideoInfo` num_frames.
     ///
     /// The `'error` lifetime is unbounded because this function always returns owned data.
     ///
     /// # Panics
     /// Panics is `n` is greater than `i32::max_value()`.
     pub fn get_frame<'error>(&self, n: usize) -> Result<FrameRef<'core>, GetFrameError<'error>> {
         assert!(n <= i32::max_value() as usize);
-        let n = n as i32;
+
+        let vi = &self.info();
+
+        #[cfg(not(feature = "gte-vapoursynth-api-32"))]
+        if let Property::Constant(total) = vi.num_frames {
+            if n >= total {
+                let err_cstring =
+                    CString::new("Requested frame number beyond the last one").unwrap();
+                return Err(GetFrameError::new(Cow::Owned(err_cstring)));
+            }
+        }
+
+        #[cfg(feature = "gte-vapoursynth-api-32")]
+        if n >= vi.num_frames {
+            let err_cstring = CString::new("Requested frame number beyond the last one").unwrap();
+            return Err(GetFrameError::new(Cow::Owned(err_cstring)));
+        }
```

Comment:
Is this really needed here? As far as I can tell, Vapoursynth does its own check: https://github.com/vapoursynth/vapoursynth/blob/d06204cab2e4fa0e352449f88d6f30c2d4ce54c9/src/cython/vapoursynth.pyx#L2075-L2085

Reply:
It is. That check only applies to Python.

I'm pretty sure we're calling https://github.com/vapoursynth/vapoursynth/blob/d06204cab2e4fa0e352449f88d6f30c2d4ce54c9/src/core/vsapi.cpp#L122
Which is always returning a pointer.

The `ArgumentTestFilter` would fail without the check.

Reply:
Actually, only `getFrameAsync` verifies that the requested index is valid.
So in a way `getFrame` is a lower level API.

Reply:
Hm, so it returns a non-null pointer and no error for invalid frame indices? Let's leave the check then, but could you report that to VapourSynth? They've been quite responsive to my past bug reports over the development of these bindings.

Reply:
I did create https://github.com/vapoursynth/vapoursynth/issues/850

Reply:
So does this mean that with variable frame count there's no way to `get_frame()` safely atm?

Reply:
> So does this mean that with variable frame count there's no way to `get_frame()` safely atm?

I don't really know how the old API worked but `num_frames` is always positive and non-zero now.